button.py

```diff
@@ -2,29 +2,70 @@
 from size import Size
 from pos import Pos
 from font import font_surface
+from text import Text
 
 import colour
 import pygame
 
 class Button(Element):
-    def __init__(self, text, font_size=11):
-        super(Button, self).__init__(Size(70, 30))
+    def __init__(self, text, font_size=11, size=Size(70, 30), pos=Pos(0, 0), background_colour=colour.WHITE):
+        super(Button, self).__init__(size, pos, background_colour)
         self.text = text
         self.font_size = font_size
-        self.border = 2
-
-    def render(self, surface):
-        abs_pos = Pos(self.origin.x + self.pos.x, self.origin.y + self.pos.y)
-        rect = (
-            abs_pos.x,
-            abs_pos.y, 
-            self.size.w, 
-            self.size.h
-        )
-        self.container = pygame.draw.rect(surface, colour.GREEN, rect)
-        surface.blit(
-            font_surface(self.text, self.font_size),
-            (abs_pos.x, abs_pos.y)
-        )
+        self.image = pygame.Surface(size)
+        self.image.fill(self.background_colour)
+        self.rect = self.image.get_rect()
+        self.rect.move_ip(pos)
```

Comment:
Are we planning to keep it as absolute positioning for this review and then change it to relative in the next iteration? I'm not sure why you got rid of the the relative positioning that seems to work before. Once the UI becomes more complex and deeply nested we don't want to compute positional offsets by hand.